com.chartboost.helium/Android~/helium-android-bridge/src/main/java/com/chartboost/heliumsdk/unity/HeliumUnityAdWrapper.kt

```diff
@@ -0,0 +1,202 @@
+
+    fun setBannerVisibility(isVisible: Boolean) {
+        HeliumUnityBridge.runTaskOnUiThread {
+            if (ad is HeliumBannerAd && bannerLayout != null) {
+                val visibility = if (isVisible) View.VISIBLE else View.INVISIBLE
+                bannerLayout?.visibility = visibility
+                ad.visibility = visibility
+            }
+        }
+    }
+
+    fun clearLoaded(): Boolean {
+        if (ad is HeliumFullscreenAd) {
+            return ad.clearLoaded()
+        }
+        if (ad is HeliumBannerAd) {
+            HeliumUnityBridge.runTaskOnUiThread { ad.clearAd() }
+            return true
+        }
+        return false
+    }
+
+    fun readyToShow(): Boolean {
+        return try {
+            when (ad) {
+                is HeliumFullscreenAd -> ad.readyToShow()
+                is HeliumBannerAd -> {
+                    Log.w(TAG, "This should never be called, banners do not have readyToShow")
+                    false
+                }
+                else -> false
+            }
+        } catch (ex: Exception) {
+            Log.w(TAG, "Helium encountered an error calling Ad.readyToShow() - ${ex.message}")
+            false
+        }
+    }
+
+    fun destroy() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            destroyBannerLayout()
+            ad.destroy()
+        }
+    }
+
+    private fun createBannerLayout(screenLocation: Int) {
+        if (ad !is HeliumBannerAd) {
+            return
+        }
+
+        if (activity == null)
+            return
+
+        // Create the banner layout on the given position.
+        // Check if there is an already existing banner layout. If so, remove it. Otherwise,
+        // create a new one.
+        var layout = bannerLayout
+
+        layout?.let {
+            it.removeAllViews()
+            val bannerParent = it.parent as ViewGroup
+            bannerParent.removeView(it)
+        }.also {
+            bannerLayout = RelativeLayout(activity)
+            layout = bannerLayout
+            layout?.setBackgroundColor(Color.TRANSPARENT)
+        }
+
+        /*
+            //     TopLeft = 0,
+            //     TopCenter = 1,
+            //     TopRight = 2,
+            //     Center = 3,
+            //     BottomLeft = 4,
+            //     BottomCenter = 5,
+            //     BottomRight = 6
+        */
+        var bannerGravityPosition = 0
+        when (screenLocation) {
+            0 -> bannerGravityPosition = Gravity.TOP or Gravity.LEFT
+            1 -> bannerGravityPosition = Gravity.TOP or Gravity.CENTER_HORIZONTAL
+            2 -> bannerGravityPosition = Gravity.TOP or Gravity.RIGHT
+            3 -> bannerGravityPosition = Gravity.CENTER_VERTICAL or Gravity.CENTER_HORIZONTAL
+            4 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.LEFT
+            5 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.CENTER_HORIZONTAL
+            6 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.RIGHT
+        }
```

Comment:
I'm pretty sure this can be rewritten so that we assign directly to the `var bannerGravityPosition`. I don't have the IDE now but it's something like:

```kt
var bannerGravityPosition = when (condition) {
    0 -> Gravity.TOP or Gravity.LEFT
    // Other cases
}
```

Reply:
Done, thanks for mentioning this, it helps improve the code quality of legacy stuff :)